### PR TITLE
[ng] ng.io sync: server-communication

### LINKED ISSUE #334: [ng] update HTTP / server-communication to at least match the tutorial HTTP page
While the tutorial section on HTTP has evolved, the server-communication has some outdated sections.

It might be a good idea to at least mark the section as outdated and to refer the reader to the tutorial HTTP section in the meantime. @kwalrath, thought?

- [x] adjust headings - the page contains more than one level-1 heading.
- [x] update prose and example to match newer material in toh-6

### PR DESCRIPTION
**Pull request by chalin**
- The main changes have been to update the sample to reflect the most recent design decisions made in the ToH, part 6.
- Note that the server-communication chapter is still incomplete -- see #260

Fixes #334
Contributes to #368 
E2E tests are passing.

Staged at: https://angulardart-org-dev2.firebaseapp.com/angular/guide/server-communication

### CODE REVIEW

File: src/angular/guide/server-communication.jade [hunk 1]
```diff
@@ -89,8 +89,8 @@ block wikipedia-jsonp+
       [example sources](https://github.com/angular-examples/server-communication)
       to see how to access Wikipedia via its `JSONP` API.
 
-block redirect-to-web-api
+//- block redirect-to-web-api
```
Comment (kwalrath) [inline src/angular/guide/server-communication.jade]: Why comment this out if the block (the whole appendix?) was removed from the TS version?

File: src/angular/guide/server-communication.jade [hunk 2]
```diff
@@ -36,20 +36,20 @@ block http-providers
...
 
-block getheroes-and-addhero
+block getheroes-and-create
   :marked
-    The hero service `getHeroes()` and `addHero()` asynchronous methods return the
+    The hero service `getHeroes()` and `create()` asynchronous methods return the
     [`Future`](https://api.dartlang.org/stable/dart-async/Future-class.html)
     values of the current hero list and the newly added hero,
-    respectively. The hero list component methods of the same name specifying
+    respectively. The hero list component `getHeroes()` and `addHero()` methods specifying
```
Comment (chalin) [inline src/angular/guide/server-communication.jade]: Oh, silly mistake. Fixed

File: src/angular/guide/server-communication.jade [hunk 3]
```diff
@@ -60,23 +60,23 @@ block getheroes-and-addhero
 block parse-json
   :marked
     The response data are in JSON string form.
-    We must parse that string into Objects which we do by calling
+    You must parse that string into objects which you do by calling
```
Comment (chalin) [inline src/angular/guide/server-communication.jade]: fixed

File: src/angular/guide/server-communication.jade [hunk 4]
```diff
@@ -60,23 +60,23 @@ block getheroes-and-addhero
...
     `!{_priv}heroService.getHeroes()` in a `try` clause. When an exception is
-    caught, the `errorMessage` variable &mdash; which we've bound conditionally in the
+    caught, the `errorMessage` variable &mdash; which you've bound conditionally in the
     template &mdash; gets assigned to.
 
 block hero-list-comp-add-hero
   :marked
-    Back in the `HeroListComponent`, we see that *its* `addHero()`
-    awaits for the *service's* asynchronous `addHero()` to return, and when it does,
+    Back in the `HeroListComponent`, you see that its `addHero()`
```
Comment (chalin) [inline src/angular/guide/server-communication.jade]: fixed

File: src/angular/guide/server-communication.jade (hunk 1 above)
Comment (chalin) [inline src/angular/guide/server-communication.jade]: Dropped it too.

**Pull request merged**